conn.go

```diff
@@ -666,6 +668,20 @@ func (cn *conn) simpleQuery(q string) (res *rows, err error) {
 	}
 }
 
+type noRows struct{}
+
+var emptyRows noRows
+
+var _ driver.Result = noRows{}
+
+func (noRows) LastInsertId() (int64, error) {
+	return 0, errors.New("no LastInsertId available after empty statement")
+}
+
+func (noRows) RowsAffected() (int64, error) {
+	return 0, errors.New("no RowsAffected available after empty statement")
```

Comment:
Exporting is probably best. Returing an error instead of just `0, nil` is consistent with https://github.com/golang/go/blob/8f9671d11a219d8fc9a6176ddf9939e743982ffc/src/database/sql/driver/driver.go#L202.